manifests/install/cni_plugins.pp

```diff
@@ -19,28 +19,50 @@
 
     '/etc/cni': ;
     '/etc/cni/net.d': ;
-    '/opt/cni': ;
+  }
+
+  file { '/opt/cni':
+    ensure  => stdlib::ensure($ensure, 'directory'),
+    purge   => true,
+    recurse => true,
+    force   => true,
   }
```

Comment:
The `/opt/cni` folder is expected to be modified by a running cluster - things like Multus, overlay network implementations, network analysis extensions, mesh components, etc will often deploy things into it. So purging it of resources that aren't Puppet-managed is very likely to cause issues.

Reply:
I wasn't aware of that. In that case we should put the downloaded files somewhere else. Maybe `/opt/cni-plugins`?

Reply:
I have moved it to `/opt/cni-plugins` in the base commit and `/opt/k8s/cni-${version}` in the second commit.